src/clients/security.rb

```diff
@@ -180,83 +179,66 @@ def SecuritySummaryHandler(options)
     def SecurityLevelHandler(options)
       options = deep_copy(options)
       current = :custom
-      Builtins.maplist(@Levels) do |key, level|
-        current = key if level == Security.Settings
-      end
-      lvl = ""
-      if options.key?("workstation")
-        lvl = "Level1"
-      elsif options.key?("roaming")
-        lvl = "Level2"
-      elsif options.key?("server")
-        lvl = "Level3"
+
+      levels = @Levels.select { |_key, level| level == Security.Settings }
+
+      current = if levels.empty?
+                  :custom
+                else
+                  levels.last # mimic the old implementation
+                end
+
+      lvl = if options.key?("workstation")
+              "Level1"
+            elsif options.key?("roaming")
+              "Level2"
+            elsif options.key?("server")
+              "Level3"
+            else
+              :custom # shouldnt :custom and levels all same type string or symbol?
       end
 
       if current != lvl
-        Security.Settings = Ops.get(@Levels, lvl, {})
+        Security.Settings = @Levels.fetch(lvl, {})
         Security.modified = true
-        return true
       end
-      false
+      Security.modified
```

Comment:
this in fact is not same as previous code as it can return true, when something else set Security.modified to true and here it is not set